lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -623,6 +658,35 @@ void LowerLayersPass::runOnModuleBody(FModuleOp moduleOp,
   });
 }
 
+void LowerLayersPass::createMacroDecls(CircuitNamespace &ns, OpBuilder &b,
+                                       LayerOp layer,
+                                       SmallVector<StringAttr> &stack) {
+  stack.emplace_back(layer.getSymNameAttr());
+  if (layer.getConvention() == LayerConvention::Inline) {
+    auto *ctx = &getContext();
+    auto macName = macroNameForLayer(stack);
+    auto symName = ns.newName(macName);
+
+    auto symNameAttr = StringAttr::get(ctx, symName);
+    auto macNameAttr = StringAttr();
+    if (macName != symName)
+      macNameAttr = StringAttr::get(ctx, macName);
+
+    b.create<sv::MacroDeclOp>(layer->getLoc(), symNameAttr, ArrayAttr(),
+                              macNameAttr);
+    macroNames[layer] = FlatSymbolRefAttr::get(&getContext(), symNameAttr);
+  }
+  for (auto child : layer.getOps<LayerOp>())
+    createMacroDecls(ns, b, child, stack);
+  stack.pop_back();
+}
+
+void LowerLayersPass::createMacroDecls(CircuitNamespace &ns, OpBuilder &b,
+                                       LayerOp layer) {
+  SmallVector<StringAttr> stack;
+  createMacroDecls(ns, b, layer, stack);
+}
```

Comment:
This function has some strict expectations on the insertion point. Also, it has one usage where the `OpBuilder` is constructed and passed in. This could save an argument and avoid the strict expectations by creating the `OpBuilder` insider this function.

```suggestion
void LowerLayersPass::createMacroDecls(CircuitNamespace &ns, LayerOp layer) {
  OpBuilder b(layer);
  SmallVector<StringAttr> stack;
  createMacroDecls(ns, b, layer, stack);
}
```